### PR TITLE
[MRG+1]: Output JUnit XML file

### PR DESCRIPTION
**Pull request by larsoner**
I thought this might be a useful feature, let's see if it actually works.

### CODE REVIEW

**Comment by larsoner** [comment]
Output is a bit underwhelming for SG, but still potentially useful esp. when there are failures:

https://circleci.com/gh/sphinx-gallery/sphinx-gallery/758

I'm going to try MNE next to see if it looks more interesting.

**Comment by larsoner** [comment]
Okay [over in MNE](https://github.com/mne-tools/mne-python/pull/6008) I got it working, too:

https://circleci.com/gh/mne-tools/mne-python/11609

This is ready for review/merge from my end.

**Comment by larsoner** [comment]
An example of a failed MNE run with more tests:

https://circleci.com/gh/mne-tools/mne-python/11613

**Comment by GaelVaroquaux** [comment]
Super nice! Should be added to the whatsnew

**Comment by larsoner** [comment]
Done

**Comment by GaelVaroquaux** [comment]
One last question: if I read this PR well, downstream needs to update their conf.py to benefit from this new awesome feature.

Is this correct?

If so, can it be avoided?

File: CHANGES.rst [hunk 1]
```diff
@@ -22,6 +22,9 @@ Developer changes
   been changed from ``url`` to ``binderhub_url`` in order to make it more
   explicit. The old configuration key (``url``) will be deprecated in
   version v0.4.0)
+* Support for generating JUnit XML summary files via the ``'junit'``
+  configuration value, which can be useful for building on CI services such as
```
Comment (GaelVaroquaux) [inline CHANGES.rst]: Maybe a link to the relevant docs, to help downstream projects?

**Comment by larsoner** [comment]
> Is this correct?

Yes

> If so, can it be avoided?

I thought about setting a default, but I didn't want to start writing files, especially since where you put them can matter in terms of CircleCI (in particular) actually recognizing them -- they want it in some subdir that you will (seemingly need to both) `store_artifacts` and `store_test_data` for. We could default to `sphinx-gallery/junit.xml` assuming people will do this for their `_build/html` dir most often, but then people who `git commit -a` blindly in deploy statements will suddenly have a `sphinx-gallery/junit.xml` file in their `github.io` or RTD outputs. But maybe that's okay/preferable.

**Comment by GaelVaroquaux** [comment]
Not sure that it's needed, because anyhow, people will need to update their circleCI.

This is great. +1 for merge from me. But please add the link in the whatsnew.

**Comment by larsoner** [comment]
Links added

**Comment by larsoner** [comment]
@choldgraf feel free to do the honors if you're happy

File: doc/configuration.rst [hunk 2]
```diff
@@ -435,6 +436,40 @@ Which is achieved by the following configuration::
...
+in the ``/build`` output directory, set the configuration key (path is relative
+to the HTML output directory)::
+
+     sphinx_gallery_conf = {
+         ...
+         'junit': '../test-results/sphinx-gallery/junit.xml',
+     }
+
+By default, JUnit XML file generation is disabled (by setting ``'junit': ''``).
+JUnit XML files be useful for example on CircleCI builds, where you can add
```
Comment (choldgraf) [inline doc/configuration.rst]:
```suggestion
JUnit XML files are useful for example on CircleCI builds, where you can add
```

**Comment by choldgraf** [comment]
one small nit from me and happy to merge. This is the first I've heard of this!

**Comment by larsoner** [comment]
I always saw that CircleCI had some JUnit functionality but had no idea how it worked. I just looked at the `pytest` output JUnit file and realized it was pretty simple. I hope it ends up being useful :)

I applied your fix and there is now +2, so I'll merge once it comes back happy. Thanks for the quick reviews!

**Pull request merged**